dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Unpickler.cs

```diff
@@ -93,24 +70,34 @@ public class Unpickler : IDisposable {
 	 * @return the reconstituted object hierarchy specified in the file.
 	 */
 	public object load(Stream stream) {
-		input = stream;
-		stack = new UnpickleStack();
-		while (true) {
-			byte key = PickleUtils.readbyte(input);
-			object value = dispatch(key);
-			if(value != NO_RETURN_VALUE)
-				return value;
-		}
-	}
+        stack = new UnpickleStack();
+        var unpickler = new Unpickler<StreamReader>(new StreamReader(stream), memo, stack, this);
+        return unpickler.Load();
+    }
 
-	/**
-	 * Read a pickled object representation from the given pickle data bytes.
-	 * 
-	 * @return the reconstituted object hierarchy specified in the file.
-	 */
-	public object loads(byte[] pickledata) {
-		return load(new MemoryStream(pickledata));
-	}
+    /// <summary>
+    /// Read a pickled object representation from the given pickle data bytes.
+    /// </summary>
+    /// <param name="pickledata">Serialized pickle data.</param>
+    /// <param name="stackCapacity">Optional parameter that suggests the initial capacity of stack. The default value is 4.</param>
+    /// <returns>the reconstituted object hierarchy specified in the memory buffer.</returns>
+    public object loads(byte[] pickledata, int stackCapacity = UnpickleStack.DefaultCapacity) {
```

Comment:
Why is this method (and the next one) named `loads` when the previous is named `load`?

Reply:
It's an existing convention (however not very intuitive).

https://github.com/irmen/Pyrolite/blob/060bc3c9069cd31560b6da4f67280736fb2cdf63/dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Unpickler.cs#L111

Reply:
To reflect the Python API.